tango/test_context.py

```diff
@@ -192,34 +332,49 @@ def post_init(self):
                 "The post_init routine failed to report anything")
             self.queue.put((None, exc, None))
 
-    def generate_db_file(self, server, instance, device,
-                         tangoclass=None, properties={}):
-        """Generate a database file corresponding to the given arguments."""
-        if not tangoclass:
-            tangoclass = server
+    def append_db_file(self, server, instance, tangoclass, device_prop_info):
+        """Generate a database file corresponding to the given arguments.
+        """
+        device_names = [info["name"] for info in device_prop_info]
         # Open the file
-        with open(self.db, 'w') as f:
+        with open(self.db, "a") as f:
             f.write("/".join((server, instance, "DEVICE", tangoclass)))
-            f.write(': "' + device + '"\n')
+            for device_name in device_names:
+                f.write(': "' + device_name + '"\n')
+            f.flush()
```

Comment:
Is there a need for this flush?  I'd expect exiting the context handler to close the file which will flush any pending writes.